woocommerce/woocommerce
# [Accessibility] Trap focus inside the product gallery modal

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #43605 .
Closes #43606 .
Closes #51925 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to a product page that has multiple images. If you have populated your store with the sample product data, you can go to the `/product/hoodie/` page.
2. Open the product gallery by clicking on the magnifying glass icon.
    ![image](https://github.com/user-attachments/assets/6f92172e-4ddf-41e2-936f-ca9276c1d6dc)
3. Once the gallery modal is open, tab navigate through the buttons and verify the focus never goes outside the modal. The order of the buttons is determined by the Photoswipe pluging and is out of the scope of this PR.
4. Close the modal and verify the focus goes back to the magnifying glass icon. 
    **HEADS-UP**: This PR doesn't implement any change on how screen readers announce the gallery trigger button (magnifying icon) since the [#50493](https://github.com/woocommerce/woocommerce/pull/50493) PR has addressed those changes already.


### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
